src/lib.rs

```diff
@@ -697,6 +698,48 @@ impl io::Write for RleVec<u8> {
     fn flush(&mut self) -> io::Result<()> { Ok( () ) }
 }
 
+impl io::Read for RleVec<u8> {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        let len = cmp::min(buf.len(), self.len());
+
+        let mut buf = buf.as_mut_ptr();
+        let mut remaining = len;
+        let mut remove_until = 0;
+
+        for (i, Run{ len, value }) in self.runs().enumerate() {
+
+            let len = if len < remaining { len }
+            else if len == remaining { remove_until = i + 1; remaining }
+            else { remaining };
+            remaining -= len;
+
+            unsafe {
```

Comment:
I tried an alternative to unsafe using
```rust
for b in &mut buf[offset..offset + len] {
    *b = *value;
}
offset += len;
```
Ran a small benchmark and actually got a better performance.
Also reading into relatively small buffers can be quite expensive because of the end position adjustments in the Run vector. Maybe we should be clear about this in the docs.